### PR TITLE
Adjust FE_DGPNonparametric to not return values on the reference cell.

### LINKED ISSUE #1870: FE_DGPNonparametric::shape_* should throw an exception, as documented
#1864 documents the `FiniteElement::shape_value` function and friends, and states that elements that don't define shape functions by mapping from the reference element shouldn't return anything in these functions but instead raise an exception.

The `FE_DGPNonparametric` class is an example of such an element, but it doesn't follow this advice. Fix this.

### PR DESCRIPTION
**Pull request by bangerth**
This is because the element does not define its shape functions by mapping
from the reference cell, so evaluating anything on the reference cell
makes no sense. Fix this by throwing an exception, as discussed in
the documentation of the corresponding functions in the base class.

This fixes #1870. In reference to #1198 and #1864.

### CODE REVIEW

File: include/deal.II/fe/fe_dgp_nonparametric.h [hunk 1]
```diff
@@ -289,63 +289,81 @@ class FE_DGPNonparametric : public FiniteElement<dim,spacedim>
...
+   * of type FiniteElement::ExcUnitShapeValuesDoNotExist.
    */
   virtual double shape_value (const unsigned int i,
                               const Point<dim> &p) const;
 
   /**
-   * Return the value of the @p componentth vector component of the @p ith
-   * shape function at the point @p p. See the FiniteElement base class for
-   * more information about the semantics of this function.
+   * This function is intended to return the value of a shape function
```
Comment (davydden) [inline include/deal.II/fe/fe_dgp_nonparametric.h]: should it not be something like:  ".. the @p componentth of a shape function..." to make it different from the previous function?

Comment (davydden) [inline include/deal.II/fe/fe_dgp_nonparametric.h]: if so, same applied to `shape_grad_component` and `shape_grad_grad_component`

File: doc/news/changes.h [hunk 2]
```diff
@@ -167,6 +167,17 @@ inconvenience this causes.
 <a name="general"></a>
 <h3>General</h3>
 <ol>
+  <li> Changed: The function FE_DGPNonparametric::shape_value() and similar
+  functions in the same class returned values and derivatives of shape
+  functions on the reference cell. However, this element is not defined
+  through mapping of shape functions from the reference cell, and consequently
+  it makes no sense to ask for this information. These functions have therefore
+  be changed to throw an exception instead, as documented in
```
Comment (davydden) [inline doc/news/changes.h]: > have therefore be changed

**Comment by davydden** [comment]
otherwise looks good to me.

File: include/deal.II/fe/fe_dgp_nonparametric.h (hunk 1 above)
Comment (bangerth) [inline include/deal.II/fe/fe_dgp_nonparametric.h]: The function always throws an exception, so I didn't want to go into details of what exactly the arguments mean. It's also documented in the base class to which I refer here.

File: doc/news/changes.h (hunk 2 above)
Comment (bangerth) [inline doc/news/changes.h]: thanks -- fixed

**Comment by bangerth** [comment]
I fixed the grammar error. What do you want me to do about the documentation of the invalid function?

**Comment by davydden** [comment]
leave it as it is now, i guess it's indeed not important in this case.

**Pull request merged**